components/shared/code/libs/lib_nvm.c

```diff
@@ -378,39 +459,53 @@ static void recordWrite(const lib_nvm_entry_t entryId)
     }
 
     lib_nvm_recordHeader_S recordHeader = { 0U };
-
     lib_nvm_recordHeader_S * const currentRecord = (lib_nvm_recordHeader_S*)records[entryId].currentNvmAddr_Ptr;
-    lib_nvm_blockHeader_S * const blockHeaderCurrent = (lib_nvm_blockHeader_S * const)getBlockBaseAddress((uint32_t)data.currentPtr);
     uint16_t entrySize = lib_nvm_entries[entryId].entrySize;
 
-    if (blockHeaderCurrent != (lib_nvm_blockHeader_S * const)getBlockBaseAddress((uint32_t)data.currentPtr + sizeof(lib_nvm_recordHeader_S) + entrySize))
+#if FEATURE_IS_ENABLED(NVM_TASK)
+    if (data.rtosStarted)
+    {
+        xSemaphoreTake(data.flash_mutex, portMAX_DELAY);
+    }
+#endif
```

Comment:
yeah this just seems like a critical section to me. I don't quite get why we need a semaphore here

Reply:
Yea i think with how it eveolved thats maybe valid, unclear havent danced around it too much mentally